src/pkg/logger/logger.go

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package logger implements a log/slog based logger in Zarf.
+package logger
+
+import (
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+	"strings"
+)
+
+// Level declares each supported log level. These are 1:1 what log/slog supports by default. Info is the default level.
+type Level int
+
+// Store names for Levels
+var (
+	Debug = Level(slog.LevelDebug) // -4
+	Info  = Level(slog.LevelInfo)  // 0
+	Warn  = Level(slog.LevelWarn)  // 4
+	Error = Level(slog.LevelError) // 8
+)
+
+// validLevels is a set that provides an ergonomic way to check if a level is a member of the set.
+var validLevels = map[Level]bool{
+	Debug: true,
+	Info:  true,
+	Warn:  true,
+	Error: true,
+}
+
+// strLevels maps a string to its Level.
+var strLevels = map[string]Level{
+	"debug": Debug,
+	"info":  Info,
+	"warn":  Warn,
+	"error": Error,
+}
+
+// ParseLevel takes a string representation of a Level, ensure it exists, and then converts it into a Level.
+func ParseLevel(s string) (Level, error) {
+	k := strings.ToLower(s)
+	l, ok := strLevels[k]
+	if !ok {
+		return 0, fmt.Errorf("invalid log level: %s", k)
+	}
+	return l, nil
+}
+
+// Format declares the kind of logging handler to use. An empty Format defaults to text.
+type Format string
+
+// ToLower takes a Format string and converts it to lowercase for case-agnostic validation. Users shouldn't have to care
+// about "json" vs. "JSON" for example - they should both work.
+func (f Format) ToLower() Format {
+	return Format(strings.ToLower(string(f)))
+}
+
+// TODO(mkcp): Add dev format
+var (
+	// FormatText uses the standard slog TextHandler
+	FormatText Format = "text"
```

Comment:
What is the difference between text and dev?

Reply:
It would point towards different handlers, the default text output does not have line numbers or pretty-printed structs, but a dev readout would provide more info. Think of it like "debug" format